util-node/createLearnPages.js

```diff
@@ -8,28 +9,66 @@ function getYamlPageIdentifier(relativePath) {
     : relativePath.replace(/(\.[a-z]+)?\.(mdx|md)/, '');
 }
 
+function getIteratedPagesForLocale(
+  items,
+  localeEdgeMap,
+  locale = defaultLanguage
+) {
+  const getLearnEdgeByPageId = pageId => edge =>
+    getYamlPageIdentifier(edge.node.parent.relativePath) === pageId;
+
+  const defaultLocaleEdges = localeEdgeMap.get(locale);
+  // This adds the items to the navigation section data based on the order defined within the YAML file
+  // If the page doesn't exist it will be set as null and then removed via Array.filter()
+  const iteratedPages = iterateEdges(
+    items
+      // Iterates the items of the section and retrieve their respective edges
+      // then we transform them into pages and add to the navigation data
+      // since learnPages are language independent we will use default edges
+      .map(pageId => defaultLocaleEdges.find(getLearnEdgeByPageId(pageId)))
+      .filter(edge => edge && edge.node)
+  );
+  return iteratedPages;
+}
+
 function createLearnPages(learnEdges, yamlNavigationData) {
   const learnPages = [];
   const navigationData = {};
 
-  const getLearnEdgeByPageId = pageId => edge =>
-    getYamlPageIdentifier(edge.node.parent.relativePath) === pageId;
+  const localeEdgeMap = new Map();
+
+  learnEdges.forEach(edge => {
+    const edgeLocale = edge.node.fields.locale;
+    if (!localeEdgeMap.has(edgeLocale)) {
+      localeEdgeMap.set(edgeLocale, []);
+    }
+    const localeEdges = localeEdgeMap.get(edgeLocale);
+    localeEdgeMap.set(edgeLocale, localeEdges.concat(edge));
+  });
 
   // Handles the Navigation Data only of Learn pages
   yamlNavigationData.forEach(({ section, items }) => {
-    navigationData[section] = [];
-
-    // This adds the items to the navigation section data based on the order defined within the YAML file
-    // If the page doesn't exist it will be set as null and then removed via Array.filter()
-    const iteratedPages = iterateEdges(
-      items
-        // Iterates the items of the section and retrieve their respective edges
-        // then we transform them into pages and add to the navigation data
-        .map(pageId => learnEdges.find(getLearnEdgeByPageId(pageId)))
-        .filter(edge => edge && edge.node)
+    const iteratedPages = getIteratedPagesForLocale(
+      items,
+      localeEdgeMap,
+      defaultLanguage
     );
 
-    navigationData[section] = iteratedPages.map(mapToNavigationData);
+    localeEdgeMap.forEach((_, locale) => {
```

Comment:
I'm sure there are better ways of doing this check without it being exhaustive. This kind of feels overly complicated.

Reply:
Yup it's surely bit complicated I'll try to come up with a good readable solution for it. I'll try to come up with solution and will try to share it with you on comments itself before actually committing it. 
Bdw this is we are doing just because  for createPage we don't get the page locale at that time. Can you explain a bit about how locale is getting set here ?

Reply:
> Bdw this is we are doing just because for createPage we don't get the page locale at that time. Can you explain a bit about how locale is getting set here ?

Yeah sure, so the locale comes from the `gatsby-theme-i18n` and it actually basically re-creates all our pages with locale data, sadly the plugin executes after when we create pages, hence the second `onCreatePage`. It's just a poor design, there's no way to get around with this with Gatsby, sadly.

I wouldn't spend that much time on the specifics, as we pretty much might leave Gatsby.

Reply:
That sounds an interesting approach, curious to how it translates in the real world? Feel free to give it a try!

To be fair, don't spend too much time on this, as we're pretty much going to move from Gatsby on the long term, but still I'd love to see with what you'd come with 🚀